r/tests/testthat/test-dplyr-funcs-type.R

```diff
@@ -625,3 +625,105 @@ test_that("bad explicit type conversions with as.*()", {
     )
   )
 })
+
+test_that("structs/nested data frames/tibbles can be created", {
+  df <- tibble(regular_col1 = 1L, regular_col2 = "a")
+
+  compare_dplyr_binding(
+    .input %>%
+      transmute(
+        df_col = tibble(
```

Comment:
Things that aren't covered by these tests (as far as I can tell) that probably should be, if they aren't elsewhere:

* Adding scalar values (not building a struct from expressions)
* Trying to add values that aren't length 1 (might need to test with a test data with more than 1 row)
* adding a data.frame defined outside of mutate/transmute